src/pcrlock/pcrlock.c

```diff
@@ -1770,6 +1766,15 @@ static int event_log_add_component_dir(EventLog *el, const char *path, char **ba
         if (r < 0)
                 return log_error_errno(r, "Failed to enumerate .pcrlock files for component '%s': %m", id);
 
+        if (!*files) {
+                log_debug("Ignoring empty directory %s", path);
+                return 0;
+        }
+
+        r = event_log_add_component(el, id, &component);
+        if (r < 0)
+                return r;
+
```

Comment:
i am pretty sure we should permit empty components, i.e. track them and show their existance in their output. we should however skip them when generating a policy. or in otherwords, a check like this makes sense i think, but not here, but instead, skip over the components when we process them if they have not a single variant defined.

Reply:
I've updated the PR to try to accomplish that. I hope it's better now